src/games/strategy/triplea/delegate/BattleDelegateOrdered.java

```diff
@@ -0,0 +1,94 @@
+package games.strategy.triplea.delegate;
+
+import java.util.Collection;
+
+import games.strategy.engine.data.Territory;
+import games.strategy.engine.data.Unit;
+import games.strategy.triplea.delegate.BattleDelegate;
+import games.strategy.triplea.delegate.IBattle;
+import games.strategy.triplea.delegate.IBattle.BattleType;
+import games.strategy.util.Match;
+
+public class BattleDelegateOrdered extends BattleDelegate {
+
+  /**
+   * Called before the delegate will run.
+   */
+  @Override
+  public void start() {
+    super.start();
+    // Fight all air and bombing raids
+    for( final Territory t : m_battleTracker.getPendingBattleSites(true) ) {   // loop through bombing/air raids
+      final IBattle raid = m_battleTracker.getPendingBattle(t, true, null);             // Get air/bombing raid for current territory
```

Comment:
The existing APIs can be nonsense. 

I suggest this: ` final IBattle raid = m_battleTracker.getPendingBattle(t, true, null);` -> `m_battleTracker.getPendingBattle(territory, BattleType.AIR_RAID_BOMBING)`

Noting that there are three comments to describe what that line is doing..
"  // Fight all air and bombing raids"
"// loop through bombing/air raids"
"// Get air/bombing raid for current territory"

Part of that same problem is this api call is a not comprehensible, no way to know what the 'true' means without looking up additional code: ` m_battleTracker.getPendingBattleSites(true)`